lib/plug/subscriber.ex

```diff
@@ -20,7 +20,38 @@ defmodule ConduitAppsignal.Plug.Subscriber do
 
   def call(message, next, opts) do
     action = "#{inspect(opts[:subscriber])}.process"
-    Appsignal.Transaction.set_action(action)
+
+    tags = %{
+      source: message.source,
+      destination: message.destination,
+      user_id: message.user_id,
+      correlation_id: message.correlation_id,
+      message_id: message.message_id
+    }
+
+    environment =
+      Map.merge(
+        %{
+          content_type: message.content_type,
+          content_encoding: message.content_encoding,
+          created_by: message.created_by,
+          created_at: message.created_at
+        },
+        message.headers
+      )
+
+    custom = %{
+      assigns: message.assigns,
+      private: message.private
```

Comment:
Things in here may not be serializable. For example, tuples and functions. Also, there may be large data structures or secrets kept in here that people may not want to send to a third party.